fix(components): [select-v2] with `teleported="false"` dropdown closes abnormally

GUEThe opened the pull request:
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

fix #16653 [Component] [select-v2] Virtualized select with teleported="false": When you click on the scrollbar, the dropdown automatically closes

GUEThe commented:
[Preview](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5jb25zdCBpbml0aWFscyA9IFsnYScsICdiJywgJ2MnLCAnZCcsICdlJywgJ2YnLCAnZycsICdoJywgJ2knLCAnaiddXG5cbmNvbnN0IHZhbHVlID0gcmVmKFtdKVxuY29uc3Qgb3B0aW9ucyA9IEFycmF5LmZyb20oeyBsZW5ndGg6IDEwMDAgfSkubWFwKChfLCBpZHgpID0+ICh7XG4gIHZhbHVlOiBgT3B0aW9uJHtpZHggKyAxfWAsXG4gIGxhYmVsOiBgJHtpbml0aWFsc1tpZHggJSAxMF19JHtpZHh9YCxcbn0pKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGVsLXNlbGVjdC12MlxuICAgIHYtbW9kZWw9XCJ2YWx1ZVwiXG4gICAgZmlsdGVyYWJsZVxuICAgIDpvcHRpb25zPVwib3B0aW9uc1wiXG4gICAgcGxhY2Vob2xkZXI9XCJQbGVhc2Ugc2VsZWN0XCJcbiAgICBzdHlsZT1cIndpZHRoOiAyNDBweFwiXG4gICAgbXVsdGlwbGVcbiAgICA6dGVsZXBvcnRlZD1cImZhbHNlXCJcbiAgLz5cbjwvdGVtcGxhdGU+XG4iLCJzcmMvUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE2NzE0LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsInNyYy9lbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL3ByZXZpZXctMTY3MTQtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTE2NzE0LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGwoc3R5bGVzKVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTE2NzE0LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

iZhangHao1997 commented:
same

GUEThe commented:
@kooriookami

kooriookami commented:
It still has problem, when you click the other blank places in dropdown, like the arrow. It'll be closed.
![image](https://github.com/element-plus/element-plus/assets/38392315/1c71d0ef-8904-4992-84f0-7c6b9fb64b73)

GUEThe commented:
> It still has problem, when you click the other blank places in dropdown, like the arrow. It'll be closed. ![image](https://private-user-images.githubusercontent.com/38392315/337146346-1c71d0ef-8904-4992-84f0-7c6b9fb64b73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NjI3NjUsIm5iZiI6MTcxNzY2MjQ2NSwicGF0aCI6Ii8zODM5MjMxNS8zMzcxNDYzNDYtMWM3MWQwZWYtODkwNC00OTkyLTg0ZjAtN2M2YjlmYjY0YjczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDA4Mjc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwOGE1ZGQ4OWVjNjNmNzIwMDRiZDE2NWU2ZjMyZWYyZWNkNWQ5NWNiNzllMmI2NWQ2OGU4NTc0YTY5ZTU1MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SnlIVJWyEZ3ilWdHuyrfEm8sWB8s4VE2DEjH5T-W72k)

What about using this way to fix , Only wrapperRef can control tooltip content visble or not
![image](https://github.com/element-plus/element-plus/assets/19553651/e9aa6604-0676-4686-b52f-9d50b8f46612)

kooriookami commented:
> > It still has problem, when you click the other blank places in dropdown, like the arrow. It'll be closed. ![image](https://private-user-images.githubusercontent.com/38392315/337146346-1c71d0ef-8904-4992-84f0-7c6b9fb64b73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NjI3NjUsIm5iZiI6MTcxNzY2MjQ2NSwicGF0aCI6Ii8zODM5MjMxNS8zMzcxNDYzNDYtMWM3MWQwZWYtODkwNC00OTkyLTg0ZjAtN2M2YjlmYjY0YjczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDA4Mjc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwOGE1ZGQ4OWVjNjNmNzIwMDRiZDE2NWU2ZjMyZWYyZWNkNWQ5NWNiNzllMmI2NWQ2OGU4NTc0YTY5ZTU1MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SnlIVJWyEZ3ilWdHuyrfEm8sWB8s4VE2DEjH5T-W72k)
> 
> What about using this way to fix , Only wrapperRef can control tooltip content visble or not ![image](https://private-user-images.githubusercontent.com/19553651/337190059-5dd557d4-5ed7-4da4-a96a-186efb593abf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NjM1NjUsIm5iZiI6MTcxNzY2MzI2NSwicGF0aCI6Ii8xOTU1MzY1MS8zMzcxOTAwNTktNWRkNTU3ZDQtNWVkNy00ZGE0LWE5NmEtMTg2ZWZiNTkzYWJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDA4NDEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4ZDMzNTZhMGQzZTY5MTk2OGM0OTQwYjM0ZmU3YWNjYjIwNTM5MjQ1NWVlMGQ0YzBiOTc1MDM1NmRlN2NjNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QUiKHJuVVS349ZBQh7IHzh58wasmp8srCCdrLhaTbR0)

Why use teleported="false" will cause this problem? I think it's the root reason of it.

GUEThe commented:
> > > It still has problem, when you click the other blank places in dropdown, like the arrow. It'll be closed. ![image](https://private-user-images.githubusercontent.com/38392315/337146346-1c71d0ef-8904-4992-84f0-7c6b9fb64b73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NjI3NjUsIm5iZiI6MTcxNzY2MjQ2NSwicGF0aCI6Ii8zODM5MjMxNS8zMzcxNDYzNDYtMWM3MWQwZWYtODkwNC00OTkyLTg0ZjAtN2M2YjlmYjY0YjczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDA4Mjc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwOGE1ZGQ4OWVjNjNmNzIwMDRiZDE2NWU2ZjMyZWYyZWNkNWQ5NWNiNzllMmI2NWQ2OGU4NTc0YTY5ZTU1MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SnlIVJWyEZ3ilWdHuyrfEm8sWB8s4VE2DEjH5T-W72k)
> > 
> > 
> > What about using this way to fix , Only wrapperRef can control tooltip content visble or not ![image](https://private-user-images.githubusercontent.com/19553651/337190059-5dd557d4-5ed7-4da4-a96a-186efb593abf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NjM1NjUsIm5iZiI6MTcxNzY2MzI2NSwicGF0aCI6Ii8xOTU1MzY1MS8zMzcxOTAwNTktNWRkNTU3ZDQtNWVkNy00ZGE0LWE5NmEtMTg2ZWZiNTkzYWJmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDA4NDEwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4ZDMzNTZhMGQzZTY5MTk2OGM0OTQwYjM0ZmU3YWNjYjIwNTM5MjQ1NWVlMGQ0YzBiOTc1MDM1NmRlN2NjNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.QUiKHJuVVS349ZBQh7IHzh58wasmp8srCCdrLhaTbR0)
> 
> Why use teleported="false" will cause this problem? I think it's the root reason of it.

When teleported="false" the el-select-menu Dom insert into the selectRef Dom as descendant Dom，the click event bubble to selectRef, trigger toggleMenu function cause  el-tooltip's content invisible( dropdown is in  el-tooltip's content). So I move the toggleMenu click event to the el-tooltip's default solt. Only the el-tooltip's default solt can control tooltip's content visble or not.

kooriookami commented:
@GUEThe Try this one, it may be the problem of tooltip
![image](https://github.com/element-plus/element-plus/assets/38392315/9ac7caad-7414-441b-b2e1-95f960258403)

GUEThe commented:
> @GUEThe Try this one, it may be the problem of tooltip ![image](https://private-user-images.githubusercontent.com/38392315/337210955-9ac7caad-7414-441b-b2e1-95f960258403.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc2NzQxNjMsIm5iZiI6MTcxNzY3Mzg2MywicGF0aCI6Ii8zODM5MjMxNS8zMzcyMTA5NTUtOWFjN2NhYWQtNzQxNC00NDFiLWIyZTEtOTVmOTYwMjU4NDAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjA2VDExMzc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEyYTUzMzdlNGI1NjBiZTFkNTNmYjNlMTAzOTlkMWViMjRiOTVlMTc0Nzg2OWFkNDIwODViNDFiYmVhY2YzNTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.tP7tIguX3W-W0RVgBlAjm_Cg5oSqcvFe2YfPGo3FAd4)

This way also can solve the bug, but I think the best way to resolve is moving toggleMenu click event. Adding `@click.stop`  might cause other problem when user want to listen the click events form ` el-tooltip-content`, or someone has already done this.

GUEThe committed:
fix(components): with teleported="false" dropdown closes bug

Move toggleMenu click event to the right place wrapperRef
Only wrapperRef can control tooltip content visble or not

closed #16653

GUEThe commented:
@kooriookami Fixed unit tests

GUEThe commented:
@btea @kooriookami Needs Review

kooriookami reviewed (approved):
@btea cc

btea reviewed (approved):
Thanks!
